docs/csharp/whats-new/csharp-version-history.md

```diff
@@ -0,0 +1,124 @@
+
+- [Asynchronous members](https://msdn.microsoft.com/library/hh191443(vs.110).aspx)
+- [Caller info attributes](https://www.codeproject.com/Tips/606379/Caller-Info-Attributes-in-Csharp)
+
+The caller info attribute is pretty cool.  It lets you easily retrieve information about the context in which you&#39;re running without resorting to a ton of boilerplate reflection code.  Many developers love this feature.
+
+But async and await are the real stars of this release.  When this came out in 2012, C# changed the game again by baking asynchrony into the language as a first class participant.  If you&#39;ve ever dealt with long running operations and the implementation of webs of callbacks, you probably loved this language feature.
+
+## C# Version 6
+
+With versions 3 and 5, C# had added some pretty impressive stuff in an OO language.  (Version 2 did as well, but it was fast following Java with those language features.)  With version 6, it would go away from doing a dominant killer feature and instead release a lot of features that delighted users of the language.  Here are some of them.
+
+- [Static imports (a la Java)](http://geekswithblogs.net/BlackRabbitCoder/archive/2015/04/16/c.net-little-wonders-static-using-statements-in-c-6.aspx)
+- [Exception filters](https://www.thomaslevesque.com/2015/06/21/exception-filters-in-c-6/)
+- [Property initializers](http://geekswithblogs.net/WinAZ/archive/2015/06/30/whatrsquos-new-in-c-6.0-auto-property-initializers.aspx)
+- [Expression bodied members](https://lostechies.com/jimmybogard/2015/12/17/c-6-feature-review-expression-bodied-function-members/)
+- [Null propagator](https://davefancher.com/2014/08/14/c-6-0-null-propagation-operator/)
+- [String interpolation](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/interpolated-strings)
+- [nameof operator](https://stackoverflow.com/questions/31695900/what-is-the-purpose-of-nameof)
+- [Dictionary initializer](https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/classes-and-structs/how-to-initialize-a-dictionary-with-a-collection-initializer)
+
+Taken individually, these are all cool language features.  But if you look at them altogether, you see an interesting pattern.  In this version, C# eliminated language boilerplate to make code more terse and readable.  So for fans of clean, simple code, this language version was a huge win.
+
+Oh, and they did do one other thing along with this version, though it&#39;s not a traditional language feature, per se.  They released [Roslyn the compiler as a service](https://github.com/dotnet/roslyn).  C# now uses C# to build C#, and they let you use the compiler as part of your programming efforts.
+
+## C# Version 7
+
+Finally, we arrive at C# version 7.  This has some evolutionary and cool stuff in the vein of C# 6, but without the compiler as a service.  Here are some of the new features.
```

Comment:
This has -> This version has

end with a colon to introduce the list of features